core/repository/sail/src/main/java/org/eclipse/rdf4j/repository/sail/SailTupleQuery.java

```diff
@@ -84,4 +85,23 @@ public void evaluate(TupleQueryResultHandler handler)
 		TupleQueryResult queryResult = evaluate();
 		QueryResults.report(queryResult, handler);
 	}
+
+	@Override
+	public QueryExplainWrapper explain(QueryExplainLevel queryExplainLevel) {
+
+		TupleExpr tupleExpr = getParsedQuery().getTupleExpr();
+
+		SailConnection sailCon = getConnection().getSailConnection();
+
+		TupleExpr explainedTupleExpr = sailCon.explain(queryExplainLevel, tupleExpr, getActiveDataset(), getBindings(),
+				getIncludeInferred());
+
+		return new QueryExplainWrapper() {
+			@Override
+			public String asHumanReadbleString() {
+				return explainedTupleExpr.toString();
+			}
+		};
+
+	}
```

Comment:
This pretty much delegates to the sail connection in the same way as `eveluate()` does.